config-v-next/eslint.cjs

```diff
@@ -393,6 +394,25 @@ function createConfig(configFilePath) {
     ],
   };
 
+  if (options.onlyHardhatError) {
+    config.overrides?.push({
+      files: ["src/**/*.ts"],
+      rules: {
+        "no-restricted-syntax": [
+          "error",
+          {
+            // This is a best effor selector that forbids every throw unless it's a `new HardhatError`,
+            // or throwing a variable within a catch clause.
+            selector:
+              "ThrowStatement:not(:has(ThrowStatement > NewExpression[callee.name='HardhatError']), CatchClause ThrowStatement:has(ThrowStatement > Identifier))",
```

Comment:
should we consider HardhatPluginError as well?

Reply:
No, as official plugins shouldn’t throw those